source/kraken/tests/realtime_test.cpp

```diff
@@ -2793,14 +2793,14 @@ BOOST_FIXTURE_TEST_CASE(add_and_update_trip_to_verify_route_line_commercial_mode
     BOOST_CHECK_EQUAL(res.journeys(0).sections(0).pt_display_informations().name(), "A - F");
 
     // Update the trip recently added with destination F replaced by J
-    new_trip = ntest::make_trip_update_message("vj_new_trip", "20190101",
-                                               {
-                                                   RTStopTime("stop_point:A", "20190101T0800"_pts).added(),
-                                                   RTStopTime("stop_point:F", "20190101T0900"_pts).skipped(),
-                                                   RTStopTime("stop_point:J", "20190101T0900"_pts).added(),
-                                               },
-                                               transit_realtime::Alert_Effect::Alert_Effect_ADDITIONAL_SERVICE,
-                                               comp_uri, phy_mode_uri);
+    new_trip =
+        ntest::make_trip_update_message("vj_new_trip", "20190101",
+                                        {
+                                            RTStopTime("stop_point:A", "20190101T0800"_pts).added(),
+                                            RTStopTime("stop_point:F", "20190101T0900"_pts).skipped(),
+                                            RTStopTime("stop_point:J", "20190101T0900"_pts).added(),
+                                        },
+                                        transit_realtime::Alert_Effect::Alert_Effect_DETOUR, comp_uri, phy_mode_uri);
```

Comment:
This test not concern `Additional Trip`. This is the reason why I have changed it.
If I understand correctly, we want `Update the trip recently added with destination F replaced by J`. It seems to be a `Detour`

Reply:
Kirin keeps the memory of an added trip, and always sends it with effect=ADDITIONAL_SERVICE: https://github.com/CanalTP/kirin/blob/master/tests/integration/cots_test.py#L1074
This is how navitia keeps the effect of the trip as ADDED: erasing previous RT, trusting Kirin.

That's why I think that:
* navitia should(can) only block RT trip-add over base-schedule VJ
* existing tests shouldn't need to be updated by this PR (except of course the one designed for it in kirin_realtime_tests.py)

Reply:
Okay I understood you remark.

We just want to reject an added trip if the base-schedule VJ is empty, because it corresponds to a real time VJ. It is the only way to distinguish a real time VJ already added.
So, I'm agree with you, `existing tests shouldn't need to be updated by this PR (except of course the one designed for it in kirin_realtime_tests.py)`